src/Validation/DNSCheckValidation.php

```diff
@@ -163,6 +169,14 @@ private function validateDnsRecords($host): bool
 
         // No MX, A or AAAA DNS records
         if ($dnsRecords === []) {
+            if ($mxRecordsResult->withError()
+                && $aRecordsResult->withError()
+                && $aaaaRecordsResult->withError()
+            ) {
+                $this->error = new InvalidEmail(new UnableToGetDNSRecord(), '');
+                return false;
+            }
```

Comment:
unsure when we should return `UnableToGetDNSRecord` if one DNS lookup failed or when all failed?